packages/manager/apps/telecom/src/app/telecom/pack/xdsl/pack-xdsl.controller.js

```diff
@@ -1,211 +1,203 @@
-angular
-  .module('managerApp')
-  .controller('PackXdslCtrl', function PackXdslCtrl(
+import { PACK_XDSL } from './pack-xdsl.constant';
+
+export default class PackXdslCtrl {
+  /* @ngInject */
+  constructor(
     $q,
     $state,
     $transitions,
     $translate,
-    $stateParams,
     OvhApiPackXdsl,
     OvhApiXdsl,
     OvhApiXdslModem,
     SidebarMenu,
     smoothScroll,
     TucToast,
     TucToastError,
-    PACK_XDSL,
   ) {
-    const animTime = 1500;
-    const noModemStatus = 404;
-    const self = this;
-
-    self.loading = {
+    this.$q = $q;
+    this.$state = $state;
```

Comment:
Nice job ! A next step could be to abstract use of `$state` in the controller

Reply:
I would like, but this `$state` is used to retrieve the current state. Is it possible to resolve this with something other ?

Reply:
We could inject `$state.current` but I think here the whole logic has to been refactored (https://github.com/ovh/manager/pull/2788/files#diff-b263ceb2d2d2f6d30ac8333b437effa5R115)

Reply:
Yes, I should think about this.